my-app/src/app/core/guards/auth.guard.ts

```diff
@@ -1,23 +1,38 @@
 import { Injectable } from '@angular/core';
-import { CanActivate, ActivatedRouteSnapshot, RouterStateSnapshot, UrlTree, Router } from '@angular/router';
+import {
+  CanActivate,
+  ActivatedRouteSnapshot,
+  RouterStateSnapshot,
+  UrlTree,
+  Router,
+} from '@angular/router';
+import { Store } from '@ngrx/store';
 import { Observable } from 'rxjs';
-import { AuthService } from '../services/auth/auth.service';
+import { map, tap } from 'rxjs/operators';
+import { AppState } from '../store/app.state';
+import { authToken } from '../store/app.state';
 
 @Injectable({
-  providedIn: 'root'
+  providedIn: 'root',
 })
 export class AuthGuard implements CanActivate {
-  constructor(private authService: AuthService, private router: Router) { }
+  constructor(private store: Store<AppState>, private router: Router) {}
 
   public canActivate(
```

Comment:
it's better to add comments to all public's with some words what's going on here